src/dune/build_system.ml

```diff
@@ -363,6 +364,47 @@ let set_rule_generators ~init ~gen_rules =
   Fdecl.set t.init_rules init_rules;
   Fdecl.set t.gen_rules gen_rules
 
+let set_vcs vcs =
+  let open Fiber.O in
+  let t = t () in
+  let () = Fdecl.set t.vcs vcs in
+  match t.caching with
+  | None -> Fiber.return ()
+  | Some ({ cache = (module Caching); _ } as caching) ->
+    let+ caching =
+      let+ with_repositories =
+        let f ({ Vcs.root; _ } as vcs) =
+          let+ commit = Vcs.commit_id vcs in
+          { Cache.directory = Path.to_absolute_filename root
+          ; remote = "FIXME: not filled yet"
```

Comment:
Is this field unused right now? If yes, shall we just remove it?

Reply:
It's defined in the original specification, I don't mind dropping it as the commit id should be sufficient. On the one hand it can be useful to determine what repository we're talking about, the git sha being a bit dry. On the other hand different people might distribute the same rule with a different remote - eg. `https://github.com/mefyl/dune` versus `https://github.com/snowleopard/dune', so this could be fragile.

However this PR change three Fixmes fields into two filled fields and a single FIXME, so I don't think this should be blocking or we should include other protocol changes in it. Granted we should probably not ship such values through the protocol, I'll move the FIXME to a comment.